added ip_selector (#11)
ipinfo/rails: README.md

 In addition to the steps listed in the Installation section it is possible to configure the library with more detail. The following arguments are allowed and are described in detail below.

```ruby

> require 'ipinfo-rails/ipselector/xforwarded_ipselector'

Reviewer: ```suggestion
require 'ipinfo-rails/ip_selector/xforwarded_ip_selector'

```

Reviewer: Not applied fully - a newline was also added here, for example.

  http_client: "",
  countries: "",
  filter: "",
  ip_selector: XForwardedIPSelector,
})
```

### IP Selection Mechanism

By default, the source IP on the request is used as the input to IP geolocation.

Since the actual desired IP may be something else, the IP selection mechanism is configurable.

Here are some built-in mechanisms:

  - [DefaultIPSelector](./lib/ipinfo-rails/ip_selector/default_ip_selector.rb)
  - [XForwardedIPSelector](./lib/ipinfo-rails/ip_selector/xforwarded_ip_selector.rb)

#### Using a custom IP selector

In case a custom IP selector is required, you may implement the `IPSelectorInterface` and pass the class to `ip_selector` in config.
```ruby

> require 'custom-package/custom_ip_selector'

Reviewer: Spacing all over here.

  http_client: "",
  countries: "",
  filter: "",

>   ipselector: XForwardedIPSelector,

Reviewer: ```suggestion
  ip_selector: XForwardedIPSelector,
```

})
```

> ### IP Selection Mechanism

Reviewer: ```suggestion

### IP Selection Mechanism
```

> By default, the IP is used by ignoring the reverse proxies depending on whether we are behind a reverse proxy or not.

Reviewer: ```suggestion
By default, the source IP on the request is used as the input to IP geolocation.
```

> Since the desired IP by your system may be in other locations, the IP selection mechanism is configurable and some alternative built-in options are available.
> Using built-in IP selectors

Reviewer: ```suggestion
Since the actual desired IP may be something else, the IP selection mechanism is configurable.

Here are some built-in mechanisms:
```

>   - [DefaultIPSelector](./lib/ipinfo-rails/ipselector/default_ipselector.rb)
>   - [XForwardedIPSelector](./lib/ipinfo-rails/ipselector/xforwarded_ipselector.rb)

Reviewer: ```suggestion
- [DefaultIPSelector](./lib/ipinfo-rails/ip_selector/default_ip_selector.rb)
- [XForwardedIPSelector](./lib/ipinfo-rails/ip_selector/xforwarded_ip_selector.rb)
```

Reviewer: The indentation decrease isn't applied.

#### Using a custom IP selector

> In case a custom IP selector is required, you may implement the `IPSelectorInterface` and pass the class to `ipselector` in config.

Reviewer: ```suggestion
In case a custom IP selector is required, you may implement `IPSelectorInterface` and pass the class to `ip_selector` in config.

```

```ruby

> require 'custom-package/custom_ipselector'

Reviewer: ```suggestion
require 'custom-package/custom_ip_selector'

```

config.middleware.use(IPinfoMiddleware, {
  token: "<your_token>",

>   ipselector: CustomIPSelector,

Reviewer: ```suggestion
  ip_selector: CustomIPSelector,
```

})
```

> 
> 

Reviewer: ```suggestion
```
